cloudinit/config/cloud-init-schema.json

```diff
@@ -1845,6 +1966,28 @@
         }
       }
     },
+    "cc_users_groups": {
+      "type": "object",
+      "properties": {
+        "groups": {
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/users_groups.group_with_users"
```

Comment:
I think this makes sense to avoid the reference, I had originally thought that items might be more complex than it ended up being. I think we should reserve $defs for things that are duplicated, or where the size of the schema is so large in oneOfs that it makes it hard to see the alternative schemas defined within a certain scope. So I'll drop  /users_groups.group_with_users and make it inline, but I'd lean toward keeping users_groups.user so we can easily see that funky "default" vs <users_groups.user> schema